src/js/feed.js

```diff
@@ -1,85 +1,114 @@
+  )?.focus()
+
+  focusOutsideNestedFeed = () => asHtml(
+    document.activeElement.parentElement.closest(ariaFeed.sArticle)
+  )?.focus()
+
+  focusInsideNestedFeed = () => asHtml(
+    $(document.activeElement.closest(ariaFeed.sArticle), ariaFeed.sArticle)
+  )?.focus()
+
+  focusBeforeFeed = () => asHtml(
+    prev(document.body, ariaFeed.sFocusable, this, {})
+  )?.focus()
+
+  focusAfterFeed = () => asHtml(
+    next(document.body, ariaFeed.sFocusable, this, {})
+  )?.focus()
+
+  setAriaAttributes = () => {
+    $$(this, ariaFeed.sScopedArticle).forEach((article, idx, articles) => {
+      article.setAttribute('tabindex', '0')
+      article.setAttribute('aria-posinset', idx + 1)
+      article.setAttribute('aria-setsize', articles.length)
+      if (!(article.hasAttribute('aria-labelledby') || article.hasAttribute('aria-label')))
+        console.error(
+          '<aria-feed>: Article has no accessible name (aria-label or aria-labelledby)',
+          article
+        )
+    })
+  }
+
+  setAriaInteractions = () => {
+    this.keydownListener = on(this, 'keydown', hotkey({
+      'PageUp': halts('default propagation', _ => this.focusPreviousArticle()),
+      'PageDown': halts('default propagation', _ => this.focusNextArticle()),
+      'Ctrl+Home': halts('default propagation', _ => this.focusBeforeFeed()),
+      'Ctrl+End': halts('default propagation', _ => this.focusAfterFeed()),
+      'Alt+PageUp': halts('default', _ => this.focusOutsideNestedFeed()),
+      'Alt+PageDown': halts('default', _ => this.focusInsideNestedFeed()),
+    }))
+  }
+
+  removeAriaInteractions = () => {
+    if (this.keydownListener) {
+      off(this.keydownListener)
+      this.keydownListener = null
+    }
+  }
+
+  constructor() {
+    super()
+    this.observer = new MutationObserver(this.handleMutations.bind(this))
+    this.observer.observe(this, {
+      childList: true,
+      attributes: false,
+      characterData: false,
+      subtree: true,
     });
-
-    on(feed, "keydown", hotkey({
-        "PageUp": halts("default propagation", _ => focusPreviousArticle(feed, root.activeElement)),
-        "PageDown": halts("default propagation", _ => focusNextArticle(feed, root.activeElement)),
-        "Alt+PageUp": halts("default", _ => focusOutsideFeed(root.activeElement)),
-        "Alt+PageDown": halts("default", _ => focusInsideFeed(root.activeElement)),
-        "Ctrl+Home": halts("default propagation", _ => focusBeforeFeed(feed)),
-        "Ctrl+End": halts("default propagation", _ => focusAfterFeed(feed)),
-    }));
-});
-
-feed(document);
+  }
+
+  connectedCallback() {
+    this.setAttribute('role', 'feed')
+    this.setAttribute('aria-live', 'polite')
+    this.setAttribute('aria-busy', 'false')
+    this.setAriaAttributes()
+    this.setAriaInteractions()
+  }
+
+  disconnectedCallback() {
+    this.observer.disconnect()
+    this.removeAriaInteractions()
+  }
+
+  handleMutations(mutationsList) {
```

Comment:
Why is this not defined as an arrow function like all the other methods? That would remove the need for `.bind` in the constructor too.

Reply:
Oof, yeah good point. I'm new to MutationObserver and cribbed a little too much from MDN.